glance/api/v2/images.py

```diff
@@ -548,15 +579,41 @@ def index(self, req, marker=None, limit=None, sort_key=None,
             self.policy.enforce(req.context, 'get_images', target)
 
             images = image_repo.list(marker=marker, limit=limit,
-                                     sort_key=sort_key,
-                                     sort_dir=sort_dir,
-                                     filters=filters,
-                                     member_status=member_status)
+                                    sort_key=sort_key,
+                                    sort_dir=sort_dir,
+                                    filters=filters,
+                                    member_status=member_status)
             db_image_count = len(images)
             images = [image for image in images
-                      if api_policy.ImageAPIPolicy(req.context, image,
-                                                   self.policy
-                                                   ).check('get_image')]
+                    if api_policy.ImageAPIPolicy(req.context, image,
+                                                self.policy
+                                                ).check('get_image')]
+
+            # CUSTOM: Determine effective domain name
+            domain_name = (
+                getattr(req.context, 'domain_name', None) or
+                getattr(req.context, 'user_domain_name', None)
```

Comment:
again: imho the user_domain_name does not matter. if they scoped to a project, the project matters. if the `domain_name` is not in the token, you'll have to ask keystone for the project's domain, the token is scoped to